src/components/ui/OverViewTable.tsx

```diff
@@ -211,39 +213,51 @@ export default function OverViewTable({
   users = { ...Object.assign({}, ...usersToDisplay) }
 
   const rows: UserRow[] = Object.values(users).map((user) => {
-    const userAsParticipant = participants.find(
+    // Find participant and stage data
+    const participant = participants.find(
       (participant) => participant.userId === user.user._id,
     )
-    const userStage = stages.find(
-      (stage) => stage._id === userAsParticipant?.stage,
+    const userStage = stages.find((stage) => stage._id === participant?.stage)
+
+    // Determine the appropriate stage based on context
+    const lastPreselectionStage = stages.findLast(
+      (stage) => stage.isPreselection === "true",
     )
+    const isPostselectionForTrainees =
+      userStage?.isPreselection !== "true" && overviewType !== "trainee"
+    const participantStage = isPostselectionForTrainees
+      ? lastPreselectionStage
+      : userStage
 
-    const status = userAsParticipant?.traineeStatus
-    const coach = coaches.find((coach) =>
+    // Find the appropriate coach
+    const coachId =
       overviewType === "trainee"
-        ? coach._id === userAsParticipant?.postselectionCoachId
-        : coach._id === userAsParticipant?.preselectionCoachId,
-    )
-    const participantPhase =
-      status === "REJECTED" || status === "DROPPED_OUT"
+        ? participant?.postselectionCoachId
+        : participant?.preselectionCoachId
+    const coach = coaches.find((c) => c._id === coachId)
+
+    // Determine participant phase
+    const status = participant?.traineeStatus
+    const isCompleted = status === "GRADUATED" || isPostselectionForTrainees
```

Comment:
please use enums as much as you can. Avoid hardcoding text for checking statuses or any other of condition.